readthedocs/core/utils/__init__.py

```diff
@@ -161,7 +161,11 @@ def prepare_build(
     # At 1h exactly, the task is grabbed by another worker and re-executed,
     # even while it's still running on the original worker.
     # https://github.com/readthedocs/readthedocs.org/issues/12317
-    if project.has_feature(Feature.BUILD_NO_ACKS_LATE):
+    if (
+        project.has_feature(Feature.BUILD_NO_ACKS_LATE)
+        or project.container_time_limit
+        and project.container_time_limit >= settings.BUILD_TIME_LIMIT
+    ):
```

Comment:
Ah okay yes, this is the update this PR needed -- avoid acks late for all projects with custom time limits, not just > 1 hour time limits.

Might as well make this check exclude the default time limit though, just in case we're somehow using the default for the project time limit.

```suggestion
    if (
        project.has_feature(Feature.BUILD_NO_ACKS_LATE)
        or project.container_time_limit
        and project.container_time_limit > settings.BUILD_TIME_LIMIT
    ):
```